flink-connector-aws-dynamodb/src/test/java/org/apache/flink/streaming/connectors/dynamodb/sink/TableRequestsContainerTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.dynamodb.sink;
+
+import org.apache.flink.streaming.connectors.dynamodb.config.DynamoDbTablesConfig;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableMap;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.PutRequest;
+import software.amazon.awssdk.services.dynamodb.model.WriteRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** Tests for {@link TableRequestsContainer}. */
+public class TableRequestsContainerTest {
+    @Test
+    public void testRequestNotDeduplicatedWhenNoTableConfig() {
+        new Scenario()
+                .witItem("table", ImmutableMap.of("pk", "1", "sk", "1"))
+                .witItem("table", ImmutableMap.of("pk", "1", "sk", "1"))
+                .withDescription("number of requests without deduplication configuration")
+                .withExpectedNumberOfItems(2)
+                .runScenario();
+    }
+
+    @Test
+    public void testDeduplicatedOnCompositeKey() {
+        new Scenario()
+                .withTableConfig("tableOne", "pk", "sk")
+                .witItem("tableOne", ImmutableMap.of("pk", "1", "sk", "1"))
+                // to be deduplicated
+                .witItem("tableOne", ImmutableMap.of("pk", "1", "sk", "1"))
```

Comment:
Should we also validate that the newer item takes precedence?

Reply:
Good point I'll add test for this

Reply:
added a test